src/ScriptBuilder/Timers/TimerDefinitionReader.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace NServiceBus.Metrics.PerformanceCounters
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using Mono.Cecil;
+
+    static class TimerDefinitionReader
+    {
+        public static bool TryGetTimerDefinition(TypeDefinition type, out List<TimerDefinition> definitions)
+        {
+            definitions = new List<TimerDefinition>();
+            if (type.BaseType != null && type.BaseType.FullName == "NServiceBus.Metrics.MetricBuilder")
+            {
+                var timerAttributes = type.Fields.Select(f => f.GetSingleAttribute("NServiceBus.Metrics.TimerAttribute")).Where(c => c != null);
+
+                foreach (var timerAttribute in timerAttributes)
+                {
+                    var name = (string) timerAttribute.ConstructorArguments[0].Value;
```

Comment:
@SimonCropp for some reason I cannot read the properties at all. I tried to make the setter private and no longer readonly but that doesn't show up with Cecil. That's why I left it with the ctor arguments. Same for the meters.